packages/jest-cli/src/TestRunner.js

```diff
@@ -67,18 +75,40 @@ class TestRunner {
     return this._buildPromise;
   }
 
-  _getAllTestPaths() {
-    return this._hasteMap
-      .matchFiles(this._testRegex)
-      .then(paths => paths.filter(path => this.isTestFilePath(path)));
+  _filterTestPathsWithStats(allPaths, options) {
+    const data = {
+      total: allPaths.length,
+      stats: {},
+    };
+
+    const testCases = options && options.checkTestPathPattern
+      ? Object.keys(this._testPathCases)
+      : Object.keys(this._testPathCases).slice(0, -1);
+
+    const paths = allPaths.filter(path => {
+      return testCases.reduce((flag, key) => {
+        if (this._testPathCases[key](path)) {
+          data.stats[key] = ++data.stats[key] || 1;
+          return flag && true;
+        }
+        data.stats[key] = data.stats[key] || 0;
+        return false;
+      }, true);
+    });
+
+    data.paths = paths;
+    return data;
+  }
+
+  _getAllTestPaths(options) {
+    return this._hasteMap.build()
+      .then(hasteMap => this._filterTestPathsWithStats(Object.keys(hasteMap.files), options));
```

Comment:
It's working actually. I've changed code a little bit in `jest-cli/src/jest.js` to be able to get stats info from the TestRunner
```js
return getTestPaths(testRunner, config, patternInfo)
  .then(data => {
    if (!data.paths.length) {
       pipe.write(`${getNoTestsFoundMessage(patternInfo, config, data)}\n`);
    }
    return data.paths;
```